src/SearchParameters/index.js

```diff
@@ -704,6 +704,43 @@ SearchParameters.prototype = {
     var newState = new ( this.constructor )( this );
     fn( newState );
     return Object.freeze( newState );
+  },
+  /**
+   * Let the user set a specific value for a given parameter. Will return the
+   * same instance if the parameter is invalid or if the value is the same as the
+   * previous one.
+   * @method
+   * @param {string} parameter the parameter name
+   * @param {any} value the value to be set, must be compliant with the definition of the attribute on the object
+   * @return {SearchParameters} the updated state
+   */
+  setQueryParameter : function setParameter( parameter, value ) {
+    var k = keys( this );
+    if( k.indexOf( parameter ) === -1 ) return this;
```

Comment:
For several reasons, it would not be consistent with the behavior of the SearchParameters and more broadly of the Helper : 

 - All setters rely on mutateMe which uses the constructor to create a new instance. The constructor is made so it explicitly take the correct parameters, it results that it also filters wrong parameters.
 - I think we're better off with a system that explicitly describes the parameters of the queries. There are many of them and it's practical to have them with the code of the client/library we use. If some parameters can be left undocumented, that's not really good.

I'll make sure we keep the parameters in sync.